File: hgdb/src/main/scala/SearchInterface.scala
package com.graphbrain.hgdb


class SearchInterface(store: VertexStoreInterface) {

  def query(text: String) = {
    val id = ID.sanitize(text)
    var maxId = 0

    while (store.exists("" + maxId + "/" + id)) {
      maxId += 1
    }

    for (i <- 1 to maxId) yield "" + i + "/" + id
  }
}

File: src/main/clojure/graphbrain/db/searchinterface.clj
(ns graphbrain.db.searchinterface
  (:require [graphbrain.db.id :as id]
            [graphbrain.db.gbdb :as gb]
            [graphbrain.db.maps :as maps]))

(defn query
  [gbdb text ctxts]
  (let [id (id/sanitize text)
        can-mean (gb/pattern->edges gbdb ["r/*can_mean" id "*"] ctxts)]
    (map #(second (maps/participant-ids %)) can-mean)))
